vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher.go

```diff
@@ -207,25 +208,87 @@ func (f *controllerFetcher) getParentOfController(controllerKey ControllerKeyWit
 		return getParentOfWellKnownController(informer, controllerKey)
 	}
 
-	// TODO: cache response
-	groupVersion, err := schema.ParseGroupVersion(controllerKey.ApiVersion)
+	groupKind, err := controllerKey.groupKind()
 	if err != nil {
 		return nil, err
 	}
-	groupKind := schema.GroupKind{
-		Group: groupVersion.Group,
-		Kind:  controllerKey.Kind,
-	}
 
 	owner, err := f.getOwnerForScaleResource(groupKind, controllerKey.Namespace, controllerKey.Name)
-	if err != nil {
+	if apierrors.IsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
 		return nil, fmt.Errorf("Unhandled targetRef %s / %s / %s, last error %v",
 			controllerKey.ApiVersion, controllerKey.Kind, controllerKey.Name, err)
 	}
 
 	return owner, nil
 }
 
+func (c *ControllerKeyWithAPIVersion) groupKind() (schema.GroupKind, error) {
+	// TODO: cache response
+	groupVersion, err := schema.ParseGroupVersion(c.ApiVersion)
+	if err != nil {
+		return schema.GroupKind{}, err
+	}
+
+	groupKind := schema.GroupKind{
+		Group: groupVersion.Group,
+		Kind:  c.ControllerKey.Kind,
+	}
+
+	return groupKind, nil
+}
+
+func (f *controllerFetcher) isWellKnown(key *ControllerKeyWithAPIVersion) (bool, error) {
```

Comment:
This never returns an error. Please return only bool.

Reply:
Fixed.